config/configmapprovider/expand_test.go

```diff
@@ -39,7 +41,9 @@ func TestBaseRetrieveFailsOnRetrieve(t *testing.T) {
 func TestBaseRetrieveFailsOnGet(t *testing.T) {
 	getErr := errors.New("test error")
 	exp := NewExpand(&mockProvider{retrieved: &mockRetrieved{getErr: getErr}})
-	defer exp.Shutdown(context.Background())
+	defer func() {
+		assert.NoError(t, exp.Shutdown(context.Background()))
```

Comment:
You can just call it with: 

```golang
defer assert.NoError(t, exp.Shutdown(context.Background()))
```

Reply:
As mentioned [here](https://go.dev/tour/flowcontrol/12),

>  The deferred call's arguments are evaluated immediately, but the function call is not executed until the surrounding function returns. 

So that call would shut down the component immediately